filesystem/ubifs.ksy

```diff
@@ -0,0 +1,856 @@
+        value: mode & 0o0170000 == 0o120000
+      is_regular:
+        value: mode & 0o0170000 == 0o100000
+      is_block_device:
+        value: mode & 0o0170000 == 0o60000
+      is_dir:
+        value: mode & 0o0170000 == 0o40000
+      is_character_device:
+        value: mode & 0o0170000 == 0o20000
+      is_fifo:
+        value: mode & 0o0170000 == 0o10000
+  master_header:
+    seq:
+      - id: highest_inum
+        type: u8
+        doc: highest inode number in the committed index
+      - id: commit_number
+        -orig-id: cmt_no
+        type: u8
+        doc: commit number
+      - id: flags
+        type: u4
+        doc: various flags (%UBIFS_MST_DIRTY, etc)
+      - id: leb_log
+        -orig-id: log_lnum
+        type: u4
+        doc: start of the log
+      - id: leb_root
+        -orig-id: root_lnum
+        type: u4
+        doc: LEB number of the root indexing node
+      - id: ofs_root
+        -orig-id: root_offs
+        type: u4
+        doc: offset within @root_lnum
+      - id: root_indexing_length
+        -orig-id: root_len
+        type: u4
+        doc: root indexing node length
+      - id: leb_garbage_collect
+        -orig-id: gc_lnum
+        type: u4
+        doc: |
+          LEB reserved for garbage collection (%-1 value means the
+          LEB was not reserved and should be reserved on mount)
+      - id: leb_index_head
+        -orig-id: ihead_lnum
+        type: u4
+        doc: LEB number of index head
+      - id: ofs_ihead
+        -orig-id: ihead_offs
+        type: u4
+        doc: offset of index head
+      - id: len_index
+        -orig-id: index_size
+        type: u8
+        doc: size of index on flash
+      - id: total_free
+        type: u8
+        doc: total free space in bytes
+      - id: total_dirty
+        type: u8
+        doc: total dirty space in bytes
+      - id: total_used
+        type: u8
+        doc: total used space in bytes (includes only data LEBs)
+      - id: total_dead
+        type: u8
+        doc: total dead space in bytes (includes only data LEBs)
+      - id: total_dark
+        type: u8
+        doc: total dark space in bytes (includes only data LEBs)
+      - id: leb_lpt
+        -orig-id: lpt_lnum
+        type: u4
+        doc: LEB number of LPT root nnode
+      - id: ofs_lpt
+        -orig-id: lpt_offs
+        type: u4
+        doc: offset of LPT root nnode
+      - id: leb_nhead
+        -orig-id: nhead_lnum
+        type: u4
+        doc: LEB number of LPT head
+      - id: ofs_nhead
+        -orig-id: nhead_offs
+        type: u4
+        doc: offset of LPT head
+      - id: leb_lprops_table
+        -orig-id: ltab_lnum
+        type: u4
+        doc: LEB number of LPT's own lprops table
+      - id: ofs_ltab
+        -orig-id: ltab_offs
+        type: u4
+        doc: offset of LPT's own lprops table
+      - id: leb_save_table
+        -orig-id: lsave_lnum
+        type: u4
+        doc: LEB number of LPT's save table (big model only)
```

Comment:
They are present unconditionally, aren't they? Maybe `makes sense only for big model` then?